kaldb/src/main/proto/manager_api.proto

```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
+import "metadata.proto";
+package slack.proto.kaldb;
+option java_package = "com.slack.kaldb.proto.manager_api";
+
+// ManagerApiService is the administration API for operating on service metadata and the associated partition assignments
+service ManagerApiService {
+  // CreateService initializes a new service in the metadata store with no initial allocated capacity
+  rpc CreateServiceMetadata(CreateServiceMetadataRequest) returns (ServiceMetadata) {}
+  // GetService returns a single service metadata by name
+  rpc GetServiceMetadata(GetServiceMetadataRequest) returns (ServiceMetadata) {}
+  // UpdateService updates an existing service with new metadata
+  rpc UpdateServiceMetadata(UpdateServiceMetadataRequest) returns (ServiceMetadata) {}
+  // GetServices returns all available services from the metadata store
+  rpc ListServiceMetadata(ListServiceMetadataRequest) returns (ListServiceMetadataResponse) {}
+  // Update partition
+  rpc UpdatePartitionAssignment(UpdatePartitionAssignmentRequest) returns (UpdatePartitionAssignmentResponse) {}
+}
+
+// CreateServiceMetadataRequest represents a new service with uninitialized thoughput and partition assignments
+message CreateServiceMetadataRequest {
+  // Name of the service, must be unique
+  string name = 1;
+  // Owner information, maybe be any string
+  string owner = 2;
+}
+
+// UpdateServiceMetadataRequest represents a request to update an existing service
+message UpdateServiceMetadataRequest {
+  // Unique name of the existing service
+  string name = 1;
+  // Owner information, maybe be any string
+  string owner = 2;
+}
+
+// GetServiceMetadataRequest represents a request to fetch an existing service
+message GetServiceMetadataRequest {
+  // Must be an exact string name match
+  string name = 1;
+}
+
+// ListServiceMetadataRequest represents a request to list all existing services
+message ListServiceMetadataRequest {}
+
+// ListServiceMetadataResponse represents the response when listing all available services
+message ListServiceMetadataResponse {
+  // List of services, excluding partition assignments
+  repeated ServiceMetadata service_metadata = 1;
+}
+
+// UpdatePartitionAssignmentRequest represents a request to allocate a new set of partitions
```

Comment:
Updated the field descriptions to indicate both throughput and partition IDs are optional, and how to mark them as such in a request. Since this is the doc for the request object and not the service, there's no allocation behavior.